Initial Draft of Reports Implementations

mail2bishnoi opened the pull request:
Hi Anton,

I have finished some parts (Report Filter and Sidebar Navigation) of the development of the report and am working on the remaining parts (Data Tables and Minor tweaking).

Please review the code and suggest further improvements.

![Mobile_Sidebar](https://user-images.githubusercontent.com/41992346/118470004-e645bb00-b723-11eb-93e0-dbd929733ec8.PNG)

![report1](https://user-images.githubusercontent.com/41992346/118470010-e776e800-b723-11eb-8c33-3417da477d42.PNG)

![reports_chart](https://user-images.githubusercontent.com/41992346/118470012-e80f7e80-b723-11eb-8069-213b893b97fe.PNG)

![reports_custom_period](https://user-images.githubusercontent.com/41992346/118470016-e8a81500-b723-11eb-96a9-a9d5e1590181.PNG)

![reports_mobile](https://user-images.githubusercontent.com/41992346/118470019-e940ab80-b723-11eb-90a7-2ce6aa1ab399.PNG)

![reports_mobile_custom_period](https://user-images.githubusercontent.com/41992346/118470021-e940ab80-b723-11eb-8865-ca431382a80b.PNG)

![reports_summary](https://user-images.githubusercontent.com/41992346/118470024-e9d94200-b723-11eb-8465-9e57209b6e58.PNG)

tananaev commented:
@dkyeremeh

tananaev commented on modern/src/theme/dimensions.js:
```diff
@@ -2,5 +2,7 @@ export default {
   sidebarWidth: '28%',
-  tabletSidebarWidth: '52px' 
+  tabletSidebarWidth: '52px',
+  drawerWidth: 360,
+  tabletDrawerWidth: 320,
```
Should these be pixels? Does `drawerWidth` refer to desktop version? Should we make it explicit?

dkyeremeh replied:
the desktop version is 360. Mobile and tablet is 320

tananaev commented on modern/src/reports/ReportLayoutPage.js:
```diff
@@ -1,42 +1,151 @@
+          </ListItemIcon>
+          <ListItemText primary={route.name} />
+        </ListItem>
+      ))}
+    </List>
+  );
+
+  const drawerHeader = (
+    <> 
+      <div className={classes.drawerHeader}>
+        <IconButton 
+          className={classes.backArrowIconContainer} 
+          disableRipple>
+          <ArrowBackIcon />
+        </IconButton>
+        <Typography variant="h6" color="inherit" noWrap>
+          {t('reportTitle')}
+        </Typography> 
+      </div>
+      <Divider />
+    </>
+  );
+
+  const appBar = (
+    <AppBar position="fixed" color="inherit">
+      <Toolbar>
+        <IconButton
+          color="inherit"
+          aria-label="open drawer"
+          edge="start"
+          onClick={() => setOpenDrawer(!openDrawer)}
+          className={classes.menuButton}>
+          <MenuIcon />
+        </IconButton>
+        <Typography variant="h6" noWrap>
+          {t('reportTitle')}
+        </Typography>        
+      </Toolbar>
+    </AppBar>   
+  );
```
What are these? Should we split them into separate components?

mail2bishnoi replied:
Split into separate components.

dkyeremeh commented:
@mail2bishnoi I've gone through your work and I'm really impressed. I noticed a few things that'll need some work though.

1. The back button in the App/Top bar should take the user to the map screen. It currently does nothing
1. The App/top bar  disappears when screen width is between `864px - 1151px` <img width="1023" alt="Screenshot 2021-05-20 at 11 05 20 AM" src="https://user-images.githubusercontent.com/52928432/118969753-f5fc1400-b95c-11eb-97da-5c953529b5a1.png">

1. I noticed that the text on `Email Report` button is spanning 2 lines on mobile and ipad. I'll drop a comment on the code where you need to fix it<img width="773" alt="Screenshot 2021-05-20 at 11 05 50 AM" src="https://user-images.githubusercontent.com/52928432/118969206-53dc2c00-b95c-11eb-8874-945a716c1c5f.png">

1. The device list was empty. This could be an issue on my side since I connected my local test app to demo3.traccar.org. So Kindly verify if it's working. <img width="574" alt="Screenshot 2021-05-20 at 11 06 29 AM" src="https://user-images.githubusercontent.com/52928432/118968870-f21bc200-b95b-11eb-8dae-b0138aaa1618.png">

dkyeremeh commented on modern/src/reports/ReportLayoutPage.js:
```diff
@@ -1,42 +1,151 @@
+          <MenuIcon />
+        </IconButton>
+        <Typography variant="h6" noWrap>
+          {t('reportTitle')}
```
Can we show the title of the current page on mobile?

mail2bishnoi replied:
Yes. I'm working on this.

mail2bishnoi commented on modern/src/reports/ReportFilter.js:
```diff
@@ -56,56 +57,80 @@ const ReportFilter = ({ children, handleSubmit, showOnly }) => {
-        </ButtonGroup>
-      </FormControl>
-    </>
+      <Grid item xs={4} sm={!showOnly ? 2 : 4}>
```
Done.

mail2bishnoi commented on modern/src/reports/ReportFilter.js:
```diff
@@ -56,56 +57,80 @@ const ReportFilter = ({ children, handleSubmit, showOnly }) => {
-        </Select>
-      </FormControl>
-      {period === 'custom' && (
+    <Grid container spacing={3}>
```
I have tried with margins and padding but have not satisfied with my implementations. Please suggest.

dkyeremeh replied:
Material-ui allows targeting of children elements when creating styles with makeStyles. So you can do something like this 
```
const useStyles = makeStyles({
  gridContainer: {
    '&>*': {
      paddingTop: theme.spacing(1.5),
      paddingBottom: theme.spacing(1.5)
  }
});
```
This will allow you to target the padding of the immediate children

mail2bishnoi commented:
Hi @tananaev @dkyeremeh 

I have implemented almost all of the reports. I am facing some problems mentioned below and need your guidance.

1. Horizontal scrollbar on DataGrid when we see reports on mobile and tablets.
2. Facing problem in making Chart responsive according to screen size.

I think the rest of the things working fine.

Thanks

tananaev commented:
@dkyeremeh can you help @mail2bishnoi. Also, do you have any designs for other features ready for us to work on?

dkyeremeh commented:
Will take a look shortly.

On Tue, 25 May 2021, 19:51 Anton Tananaev ***@***.***> wrote:

> @dkyeremeh <https://github.com/dkyeremeh> can you help @mail2bishnoi
> <https://github.com/mail2bishnoi>. Also, do you have any designs for
> other features ready for us to work on?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/traccar/traccar-web/pull/852#issuecomment-848215166>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMTZ7METM5V2F5O45H6GYYLTPP5UNANCNFSM45ADWKEA>
> .
>

-- 
The information transmitted by this email is intended only for the person 
or entity to which it is addressed. The information in the email may 
include trade secrets, privileged, or otherwise confidential information 
(including pricing or contract terms). Unauthorized review, forwarding, 
printing, copying, distributing, or using such information is strictly 
prohibited and may be unlawful. If you received this in error, please 
contact the sender and delete the material from all computers.

dkyeremeh commented:
@tananaev I've done all the designs. Will create tasks for them shortly

dkyeremeh commented:
@mail2bishnoi Due to the design of the datagrid component, applying responsiveness will not be straight-forward. But this is how to accomplish it. 

- We will only be changing the style for down('sm')
- You'll need to put the table/chart in a div wrapper. The wrapper should fill the width of the parent. So it will be `calc(100vw - ${theme.spacing(6))}`. Then set `overflow-x: "auto"`
- You'll also need to set a width for the table depending on the number of columns. The rule of thumb is that the width should allow the text of table header to show fully. 
- A size of 900px should work for most tables. Since the summary table has more columns, we may need to make it wider. I think 1024px should work

dkyeremeh commented:
@mail2bishnoi One more thing... Kindly add a padding of 16px to the top of the navigation drawer

![Uploading Screenshot 2021-05-26 at 12.14.56 PM.png…]()

mail2bishnoi commented:
Hi @tananaev @dkyeremeh 

I have almost completed all the reports.

In all the reports DataGrid, I have added width and defined the width in the dimension file. Width is defined according to the type of the column e.g. DateColumn, StringColumn, NumberColumn, and BooleanColumn, etc.  I think this approach is better than defining fixed-width container because according to screen size and column width DataGrid automatically add a scrollbar and its working fine on all the screens.

In the Charts report, when I add width="100%", then the chart occupies the entire screen width as attached below. To work around this I added withWidth() and it's working fine.

![chart_full_screen_issue](https://user-images.githubusercontent.com/41992346/120168818-ed3cf500-c21c-11eb-86c1-fc55623f691c.PNG)


Thanks.

mail2bishnoi commented:
I might not work for the next few days because I have mild covid symptoms since today morning. When I'll recover or feel better, I'll pick up a task from the list and inform you.

Thanks.

dkyeremeh commented:
Your implementation sounds good. I'll take a look soon.

dkyeremeh commented:
@mail2bishnoi sorry to hear that you're not feeling well. Hope you recover
soon

On Mon, 31 May 2021, 09:05 Ashutosh Bishnoi ***@***.***>
wrote:

> I might not work for the next few days because I have mild covid symptoms
> since today morning. When I'll recover or feel better, I'll pick up a task
> from the list and inform you.
>
> Thanks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/traccar/traccar-web/pull/852#issuecomment-851342176>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMTZ7MFHJUTQB5P6XFW2B6DTQNGN3ANCNFSM45ADWKEA>
> .
>

-- 
The information transmitted by this email is intended only for the person 
or entity to which it is addressed. The information in the email may 
include trade secrets, privileged, or otherwise confidential information 
(including pricing or contract terms). Unauthorized review, forwarding, 
printing, copying, distributing, or using such information is strictly 
prohibited and may be unlawful. If you received this in error, please 
contact the sender and delete the material from all computers.

mail2bishnoi commented:
Hi @tananaev @dkyeremeh 

I'm back to work.

I would like to finish this PR before switching to other tasks. Can you please review this for merging?

tananaev commented on modern/src/SocketController.js:
```diff
@@ -27,15 +27,18 @@ const SocketController = () => {
   const authenticated = useSelector(state => !!state.session.user);
 
   const connectSocket = () => {
+    console.log('connect socket method invoked');
```
Please remove logging.

mail2bishnoi replied:
Done.

tananaev commented on modern/src/components/reports/ReportNavbar.js:
```diff
@@ -0,0 +1,26 @@
+          <MenuIcon />
+        </IconButton>
+        <Typography variant="h6" noWrap>
+          {t('reportTitle')} {reportTitle ? `/ ${reportTitle}` : ''}
```
Is there a report without title?

mail2bishnoi replied:
No. Fixed it.

tananaev commented on modern/src/components/reports/ReportNavbar.js:
```diff
@@ -0,0 +1,26 @@
+          color="inherit"
+          aria-label="open drawer"
+          edge="start"
+          onClick={() => setOpenDrawer(!openDrawer)}>
```
This can probably just be a callback. You don't need to pass `openDrawer`.

mail2bishnoi replied:
Ok. Removed openDrawer.

tananaev commented on modern/src/components/reports/ReportSidebar.js:
```diff
@@ -0,0 +1,29 @@
+import { List, ListItem, ListItemText, ListItemIcon } from '@material-ui/core';
+import { Link, useLocation } from 'react-router-dom';
+
+const ReportNavbar = ({ routes }) => {
```
Why do you have 2 `ReportNavbar`s?

mail2bishnoi replied:
Its typo. Its ReportSidebar.

tananaev commented on modern/src/reports/ReportFilter.js:
```diff
@@ -56,56 +68,80 @@ const ReportFilter = ({ children, handleSubmit, showOnly }) => {
+        </Button>
+      </Grid>
+      <Grid item xs={4} sm={2}>
+        {!showOnly && 
```
This check can probably be moved up.

mail2bishnoi replied:
Done.

tananaev commented on modern/src/reports/ReportFilter.js:
```diff
@@ -56,56 +68,80 @@ const ReportFilter = ({ children, handleSubmit, showOnly }) => {
+          </Button>}
+      </Grid>
+      <Grid item xs={4} sm={2}>
+        {!showOnly && 
```
Same here.

mail2bishnoi replied:
Done.

tananaev commented on modern/src/reports/StopReportPage.js:
```diff
@@ -37,43 +40,43 @@ const StopReportPage = () => {
     field: 'startTime',
     type: 'dateTime',
-    flex: 1,
+    width: theme.dimensions.dateColumnWidth,
```
Not sure I'm very happy with fixed width for columns, but we can leave it as is for now.

mail2bishnoi replied:
ok.

tananaev commented on modern/src/theme/dimensions.js:
```diff
@@ -2,5 +2,11 @@ export default {
   borderRadius: '4px',
   sidebarWidth: '28%',
-  tabletSidebarWidth: '52px' 
+  tabletSidebarWidth: '52px',
+  desktopDrawerWidth: '360px',
+  tabletDrawerWidth: '320px',
```
Can we move screen size from prefix to suffix. This way it's easier to read. For example
```
drawerWidthDesktop
drawerWidthTablet
```

mail2bishnoi replied:
Done.

tananaev commented on modern/src/theme/dimensions.js:
```diff
@@ -2,5 +2,11 @@ export default {
+  tabletSidebarWidth: '52px',
+  desktopDrawerWidth: '360px',
+  tabletDrawerWidth: '320px',
+  dateColumnWidth: 160,
+  numberColumnWidth: 130,
+  stringColumnWidth: 160,
+  booleanColumnWidth: 130
```
Same here.

mail2bishnoi replied:
Done.

tananaev commented:
@mail2bishnoi Can you please rebase to resolve merge conflict.

@dkyeremeh Does everything else look good to you?

dkyeremeh commented:
Yh @tananaev looks good to me

mail2bishnoi commented:
Done.

tananaev commented:
Merged, thank you.

tananaev commented:
@mail2bishnoi when I open reports in mobile or tablet view there is no back button for some reason. Is it just me or something is broken.

@dkyeremeh what do you think about bringing back "brand" color to reports? I think without blue toolbar the screen looks a bit inconsistent with the rest of the app.
